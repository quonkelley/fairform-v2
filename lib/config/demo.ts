/**
 * Firebase Configuration Utilities
 *
 * Provides functions for returning Firebase configuration.
 */

export interface FirebaseConfig {
  apiKey: string | undefined;
  authDomain: string | undefined;
  projectId: string | undefined;
  storageBucket: string | undefined;
  messagingSenderId: string | undefined;
  appId: string | undefined;
}

/**
 * Returns the Firebase configuration
 */
export function getFirebaseConfig(): FirebaseConfig {
  return {
    projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
    apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
    authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN,
    storageBucket: process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET,
    messagingSenderId: process.env.NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID,
    appId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID,
  };
}

/**
 * Returns the Firebase Admin configuration
 * Used for server-side Firebase Admin SDK initialization
 */
export function getFirebaseAdminConfig() {
  return {
    projectId: process.env.FIREBASE_PROJECT_ID,
    clientEmail: process.env.FIREBASE_CLIENT_EMAIL,
    privateKey: process.env.FIREBASE_PRIVATE_KEY?.replace(/\\n/g, '\n'),
  };
}

/**
 * Validates that the required Firebase configuration is present
 * Throws an error with helpful message if configuration is missing
 */
export function validateFirebaseConfig(config: FirebaseConfig): void {
  const requiredFields: (keyof FirebaseConfig)[] = [
    'apiKey',
    'authDomain',
    'projectId',
    'storageBucket',
    'messagingSenderId',
    'appId',
  ];

  const missingFields = requiredFields.filter((field) => !config[field]);

  if (missingFields.length > 0) {
    throw new Error(
      `Missing required Firebase configuration: ${missingFields.join(', ')}. ` +
      `Please check your environment variables.`
    );
  }
}

/**
 * Validates that the required Firebase Admin configuration is present
 * Throws an error with helpful message if configuration is missing
 */
export function validateFirebaseAdminConfig(config: {
  projectId: string | undefined;
  clientEmail: string | undefined;
  privateKey: string | undefined;
}): void {
  if (!config.projectId || !config.clientEmail || !config.privateKey) {
    const missing: string[] = [];
    if (!config.projectId) missing.push('projectId');
    if (!config.clientEmail) missing.push('clientEmail');
    if (!config.privateKey) missing.push('privateKey');

    throw new Error(
      `Missing required Firebase Admin configuration: ${missing.join(', ')}. ` +
      `Please check your environment variables.`
    );
  }
}
