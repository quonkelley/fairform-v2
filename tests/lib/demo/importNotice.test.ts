import { describe, it, expect, vi, beforeEach } from 'vitest';
import { parseDemoNotice, getAvailableNoticeFiles, isNoticeFileSupported } from '@/lib/demo/importNotice';

// Mock the demo config timing
vi.mock('@/lib/demo/demoConfig', () => ({
  demoConfig: {
    timing: {
      scanDelay: 0, // No delay in tests
    },
  },
}));

describe('parseDemoNotice', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  describe('Happy Path', () => {
    it('should successfully parse eviction notice fixture', async () => {
      const result = await parseDemoNotice('eviction.notice.json');
      
      expect(result.success).toBe(true);
      if (result.success) {
        expect(result.data.case.caseType).toBe('eviction');
        expect(result.data.case.caseNumber).toBe('49K01-2510-EV-001234');
        expect(result.data.case.jurisdiction).toBe('Marion County, IN - Center Township Small Claims Court');
        expect(result.data.case.title).toBe('Eviction Defense - Marion County');
        expect(result.data.case.parties.plaintiff).toBe('ABC Property Management LLC');
        expect(result.data.case.parties.defendant).toBe('Alex Rodriguez');
        expect(result.data.case.propertyAddress).toBe('1234 E. 38th Street, Indianapolis, IN 46205');
        expect(result.data.case.rentOwed).toBe('$1,200.00');
        expect(result.data.case.noticeDescription).toBe('10-day notice to pay or quit');
        
        // Verify court info
        expect(result.data.case.courtInfo.code).toBe('K01');
        expect(result.data.case.courtInfo.name).toBe('Center Township Small Claims Court');
        expect(result.data.case.courtInfo.address).toBe('7201 E. 75th Street, Indianapolis, IN 46256');
        expect(result.data.case.courtInfo.phone).toBe('(317) 595-5000');
        expect(result.data.case.courtInfo.hours).toBe('Monday-Friday 8:00 AM - 4:30 PM');
        
        // Verify timeline
        expect(result.data.timeline.noticeDate).toEqual(new Date('2025-10-10'));
        expect(result.data.timeline.responseDueDate).toEqual(new Date('2025-10-20'));
        expect(result.data.timeline.hearingDate).toEqual(new Date('2025-11-06'));
        
        // Verify glossary keys
        expect(result.data.glossaryKeys).toEqual(['eviction-notice', 'answer', 'default-judgment', 'habitability']);
        
        // Verify metadata
        expect(result.data.metadata.source).toBe('Marion County Civil Data Access Research');
        expect(result.data.metadata.filingFee).toBe('$86.00');
        expect(result.data.metadata.serviceFee).toBe('$28.00');
      }
    });

    it('should include detailed notes with eviction information', async () => {
      const result = await parseDemoNotice('eviction.notice.json');
      
      expect(result.success).toBe(true);
      if (result.success) {
        expect(result.data.case.notes).toContain('Case created from uploaded eviction notice');
        expect(result.data.case.notes).toContain('Notice Type: 10-day notice to pay or quit');
        expect(result.data.case.notes).toContain('Date Received: 2025-10-10');
        expect(result.data.case.notes).toContain('Location: 1234 E. 38th Street, Indianapolis, IN 46205');
        expect(result.data.case.notes).toContain('Court: Center Township Small Claims Court');
        expect(result.data.case.notes).toContain('Case Number: 49K01-2510-EV-001234');
        expect(result.data.case.notes).toContain('Plaintiff: ABC Property Management LLC');
        expect(result.data.case.notes).toContain('Defendant: Alex Rodriguez');
        expect(result.data.case.notes).toContain('Rent Owed: $1,200.00');
        expect(result.data.case.notes).toContain('Filing Fee: $86.00');
        expect(result.data.case.notes).toContain('Service Fee: $28.00');
      }
    });
  });

  describe('Error Handling', () => {
    it('should return error for unknown filename', async () => {
      const result = await parseDemoNotice('unknown.notice.json');
      
      expect(result.success).toBe(false);
      if (!result.success) {
        expect(result.error).toBe("Notice file 'unknown.notice.json' not found");
        expect(result.code).toBe('UNKNOWN_FILE');
      }
    });
  });

  describe('Utility Functions', () => {
    it('should return available notice files', () => {
      const files = getAvailableNoticeFiles();
      expect(files).toContain('eviction.notice.json');
    });

    it('should check if notice file is supported', () => {
      expect(isNoticeFileSupported('eviction.notice.json')).toBe(true);
      expect(isNoticeFileSupported('unknown.notice.json')).toBe(false);
    });
  });

  describe('Snapshot Tests', () => {
    it('should match snapshot for successful parsing result', async () => {
      const result = await parseDemoNotice('eviction.notice.json');
      
      expect(result.success).toBe(true);
      if (result.success) {
        expect(result.data).toMatchSnapshot();
      }
    });

    it('should match snapshot for error result', async () => {
      const result = await parseDemoNotice('unknown.notice.json');
      
      expect(result.success).toBe(false);
      if (!result.success) {
        expect(result).toMatchSnapshot();
      }
    });
  });
});