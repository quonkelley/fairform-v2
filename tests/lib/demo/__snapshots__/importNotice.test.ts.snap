// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseDemoNotice > Snapshot Tests > should match snapshot for error result 1`] = `
{
  "code": "UNKNOWN_FILE",
  "error": "Notice file 'unknown.notice.json' not found",
  "success": false,
}
`;

exports[`parseDemoNotice > Snapshot Tests > should match snapshot for successful parsing result 1`] = `
{
  "case": {
    "caseNumber": "49K01-2510-EV-001234",
    "caseType": "eviction",
    "courtInfo": {
      "address": "7201 E. 75th Street, Indianapolis, IN 46256",
      "code": "K01",
      "hours": "Monday-Friday 8:00 AM - 4:30 PM",
      "name": "Center Township Small Claims Court",
      "phone": "(317) 595-5000",
    },
    "jurisdiction": "Marion County, IN - Center Township Small Claims Court",
    "notes": "Case created from uploaded eviction notice.

**Eviction Details:**
- Notice Type: 10-day notice to pay or quit
- Date Received: 2025-10-10
- Location: 1234 E. 38th Street, Indianapolis, IN 46205
- Court: Center Township Small Claims Court
- Case Number: 49K01-2510-EV-001234

**Parties:**
- Plaintiff: ABC Property Management LLC
- Defendant: Alex Rodriguez

**Financial:**
- Rent Owed: $1,200.00
- Filing Fee: $86.00
- Service Fee: $28.00",
    "noticeDescription": "10-day notice to pay or quit",
    "parties": {
      "defendant": "Alex Rodriguez",
      "plaintiff": "ABC Property Management LLC",
    },
    "propertyAddress": "1234 E. 38th Street, Indianapolis, IN 46205",
    "rentOwed": "$1,200.00",
    "title": "Eviction Defense - Marion County",
  },
  "glossaryKeys": [
    "eviction-notice",
    "answer",
    "default-judgment",
    "habitability",
  ],
  "metadata": {
    "filingFee": "$86.00",
    "serviceFee": "$28.00",
    "source": "Marion County Civil Data Access Research",
  },
  "timeline": {
    "hearingDate": 2025-11-06T00:00:00.000Z,
    "noticeDate": 2025-10-10T00:00:00.000Z,
    "responseDueDate": 2025-10-20T00:00:00.000Z,
  },
}
`;
